### PR TITLE
add missing factories to curve models

### PR DESCRIPTION
**Pull request by viniabussafi**
### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [x] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** https://github.com/duneanalytics/spellbook/pull/6218 added a missing factory to curvefi_ethereum.view_pools. Now, this PR adds it to curve_ethereum_base_trades, in order to also update dex.trades. It also adds the same factory to optimism to curvefi_optimism_pools

### CODE REVIEW

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @viniabussafi

**Comment by viniabussafi** [comment]
hey @Hosuke, can you help out with this error please? I wasn't getting it until last week (you even approved this PR). I'm not sure what may be causing this

**Review by Hosuke — COMMENTED** [review]
The problem may be caused by this section:
```
        , case
            when l.topic0 = 0x8b3e96f2b889fa771c53c981b40daf005f63f637f1869f707052d15a3dd97140
                then p.coins[cast(bytearray_to_uint256(bytearray_substring(l.data, 65, 32)) as int) + 1] 
                else p.undercoins[cast(bytearray_to_uint256(bytearray_substring(l.data, 65, 32)) as int) + 1]
            end as token_bought_address
        , case
            when l.topic0 = 0x8b3e96f2b889fa771c53c981b40daf005f63f637f1869f707052d15a3dd97140
                then p.coins[cast(bytearray_to_uint256(bytearray_substring(l.data, 1, 32)) as int) + 1] 
                else p.undercoins[cast(bytearray_to_uint256(bytearray_substring(l.data, 1, 32)) as int) + 1]
            end as token_sold_address
```

We may need to check the actual length of these fields, 
or we can do something like:
```
CASE 
    WHEN cast(bytearray_to_uint256(bytearray_substring(l.data, 65, 32)) as int) + 1 <= CARDINALITY(p.coins)
    THEN p.coins[cast(bytearray_to_uint256(bytearray_substring(l.data, 65, 32)) as int) + 1]
    ELSE NULL
END as token_bought_address
```

**Comment by viniabussafi** [comment]
your suggestions seems to work @Hosuke! Looks like we're good to go now. Thank you very much once again

**Comment by jeff-dude** [comment]
@Hosuke give this one last glance after latest changes, then i can help get it moving along if approved

File: dex/models/trades/ethereum/platforms/curvefi_ethereum_base_trades.sql [hunk 1]
```diff
@@ -31,21 +31,29 @@ WITH dexs AS
         , cast(NULL as double) AS amount_usd
         , case
             when l.topic0 = 0x8b3e96f2b889fa771c53c981b40daf005f63f637f1869f707052d15a3dd97140
+                    and cast(bytearray_to_uint256(bytearray_substring(l.data, 65, 32)) as int) + 1 <= CARDINALITY(p.coins)
                 then p.coins[cast(bytearray_to_uint256(bytearray_substring(l.data, 65, 32)) as int) + 1] 
-                else p.undercoins[cast(bytearray_to_uint256(bytearray_substring(l.data, 65, 32)) as int) + 1]
+            when l.topic0 != 0x8b3e96f2b889fa771c53c981b40daf005f63f637f1869f707052d15a3dd97140
+                    and cast(bytearray_to_uint256(bytearray_substring(l.data, 65, 32)) as int) + 1 <= CARDINALITY(p.undercoins)    
+                then p.undercoins[cast(bytearray_to_uint256(bytearray_substring(l.data, 65, 32)) as int) + 1]
+                else NULL
```
Comment (Hosuke) [inline dex/models/trades/ethereum/platforms/curvefi_ethereum_base_trades.sql:39]: `NULL` value may be introduced here.

**Review by Hosuke — APPROVED** [review]
LGTM.✅
Thank you @viniabussafi

**Comment by Hosuke** [comment]
Here are records with null token_*_address:
https://dune.com/queries/3916398
All of them (13 in total) with version `Factory V1 Stableswap Plain NG`.

**Comment by viniabussafi** [comment]
hey, @jeff-dude! can we please move forward with this still this week? would be really helpful to have the trades spell complete for curve

**Pull request merged**